lessons/basics/modules.md

```diff
@@ -249,8 +288,65 @@ defmacro __using__(_) do
 end
 ```
 
-The `Ecto.Migration.__using__/1` macro includes an import call so that when you `use Ecto.Migration` you also `import Ecto.Migration`. It also sets up a module property which we will assume controls Ecto’s behavior.
+The `Ecto.Migration.__using__/1` macro includes an import call so that when you `use Ecto.Migration` you also `import Ecto.Migration`.
+It also sets up a module property which we will assume controls Ecto’s behavior.
+
+We can also override a given function that is given to use with `__using__/1`.
+This is done by using `defoverridable`.
+Let's implement this:
+
+```elixir
+iex> defmodule Foo do
```

Comment:
Traditionally we've not put module code within `iex` because it becomes more difficult to read.

Let's try to keep the lesson styles the same